### DISCUSSION TITLE
Bluetooth Spoofing to unmodded Switch?

### DISCUSSION THREAD

**Discussion by Fan8099** [ndeadly/MissionControl#895] [Q&A]
Is it possible to use the Bluetooth spoofing feature in such a way that controllers can seamlessly connect to two different Switches (one of them being unmodded) without having to re-pair them, and without a Link Key? The best I've gotten so far is my 8bitdo Pro 2 (S Mode) being able to automatically pair without holding down the Sync button (but still needing to go to the "Change Grip/Order" screen). Joy-Con seem to try to connect to the other Switch, but turn off once they find it (no, this is not them turning off because of the timeout, they turn off much sooner than that).

**Comment by ndeadly** [comment]
In general, no. One way or another you need to be running homebrew to either read or write the pairing data on both consoles. What _might_ be possible however is pairing _official_ controllers with the unmodded console, and extracting the pairing information from the controller memory instead for injection on the modded console. This won't work in the general case though, as pairing information can't be read  from most controllers (or at least not in some standardised way)

**Reply by Fan8099** [reply]
What pairing data is needed exactly? I was able to read the Bluetooth MAC Address of the unmodded console, if that helps (not sure what host_name is, though).

Also, would I be able to use the spoofing feature for other devices, such as for phones or computers? Or is a Link Key always required?

**Reply by ndeadly** [reply]
Just the host (console) address and link key as far as i know. The locations in controller memory can be found [here](https://github.com/dekuNukem/Nintendo_Switch_Reverse_Engineering/blob/master/spi_flash_notes.md#x2000-pairing-info). You should be able to use the spoofing feature with any device as long as you can read those two fields from the primary host and inject them into the database on the secondary one. Spoofing just makes it so that the address used by your switch its changed to match the other device. You will always require the link key though, so at least one of those devices either needs to be hacked or open so that you can extract it.
